### PR TITLE
Views

### PR DESCRIPTION
**Pull request by Ellon-M**
## PR Changes :tennis: 

- Used preprocessed HTML file with embedded Ruby code.
- Used layouts and templates for shared content.
- Added CSS styles for HTML elements.

### CODE REVIEW

File: app/views/users/show.html.erb [hunk 1]
```diff
@@ -1 +1,39 @@
...
+            <span class="likes">Likes:<%= post.likes_counter || 0 %></span>
+          </div>
+          </a>
+        </li>
+      <% end %>
+    </ul>
+  </div>
+
+  <a class="see-all-posts-link" href="<%= @user.id %>/posts">See all posts</a>
+
```
Comment (brytebee) [inline app/views/users/show.html.erb:38]: This button throws an error kindly fix it.

image

File: app/views/users/show.html.erb [hunk 2]
```diff
@@ -1 +1,39 @@
...
+  <div class="user-info-bio-div">
+    <h3>Bio</h3>
+    <p><%= @user.bio%></p>
+  </div>
+
+  <div class="user-posts-cont">
+    <ul class="user-post-list">
+      <% @user.recent_posts.each_with_index do |post, i| %>
+        <li class="user-post">
+          <a href="<%=@user.id%>/posts/<%=post.id%>">
```
Comment (brytebee) [inline app/views/users/show.html.erb:24]: With this link you are required to `show` the post details, however, it throws the error below.

image

File: db/schema.rb [hunk 3]
```diff
@@ -34,22 +34,24 @@
   end
 
   create_table "posts", force: :cascade do |t|
-    t.bigint "author_id", null: false
     t.string "title"
     t.string "text"
     t.string "bio"
```
Comment (brytebee) [inline db/schema.rb:39]: - [ ] **Optional:** `bio` is not a required column for the post model/table.

File: app/views/posts/index.html.erb [hunk 4]
```diff
@@ -1 +1,39 @@
...
+            <span class="comments">Comments: <%= post.comments_counter || 0 %>, </span>
+            <span class="likes">Likes: <%= post.likes_counter || 0 %></span>
+          </div>
+        </div>
+      </a>
+
+      <ul class="comments-list">
+        Comments:
+        <div class="comment-group">
+          <% post.recent_comments.each do |comment| %>
```
Comment (brytebee) [inline app/views/posts/index.html.erb:31]: This is likely to throw errors as you have a naming conflict. In the Post model, you have named this method `most_recent_comments` as shown below. Kindly look into it.

https://github.com/Ellon-M/ror-blog-app/blob/0243444a90e9f80e14d3e514f8549e245848a2ac/app/models/post.rb#L24-L26

**Review by brytebee — COMMENTED** [review]
Hi, @Ellon-M👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Ellon-M👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB design.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Ellon-M 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**